node/pkg/driver/device_connectivity/device_connectivity_interface.go

```diff
@@ -25,4 +25,6 @@ type OsDeviceConnectivityInterface interface {
 	FlushMultipathDevice(mpathDevice string) error
 	RemovePhysicalDevice(sysDevices []string) error
 	ValidateLun(lun int, sysDevices []string) error
+	GetVolumeIdVariations(volumeId string) []string
+	GetVolumeIdByVolumePath(volumePath string, volumeId string) (string, error)
```

Comment:
where do we call these methods?
do they have to be exported ("public")?
do we even need them here in (non-connectivity-helper) scsi generic?

Reply:
[here](https://github.com/IBM/ibm-block-csi-driver/blob/8a1076ba5e079dffac2f867a69ab797bb4477f32/node/pkg/driver/node_test.go#L54) in the tests, the `newTestNodeService` function gets [`device_connectivity.OsDeviceConnectivityInterface`](https://github.com/IBM/ibm-block-csi-driver/blob/develop/node/pkg/driver/device_connectivity/device_connectivity_interface.go) and uses it's variable to pass it [to pass it to](https://github.com/IBM/ibm-block-csi-driver/blob/8a1076ba5e079dffac2f867a69ab797bb4477f32/node/pkg/driver/node_test.go#L61) [`OsDeviceConnectivityHelperScsiGenericInterface`](https://github.com/IBM/ibm-block-csi-driver/blob/7a9d7dfef1aca9ffc69211291f5dc9ced31512be/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L38) that has the functions that you marked, if I won't add them to it the tests won't run

Reply:
strange - why would `newTestNodeService` ask for `OsDeviceConnectivityInterface` when it actually needs `OsDeviceConnectivityHelperScsiGenericInterface`?
could we [mock](https://github.com/IBM/ibm-block-csi-driver/blob/8a1076ba5e079dffac2f867a69ab797bb4477f32/node/pkg/driver/node_test.go#L776) the helper instead?

Reply:
then why do I still see changes in `node/pkg/driver/device_connectivity/device_connectivity_*.go` files?

Reply:
I thought I removed all the functions from there, sorry done